pkg/storage/bloom/v1/tokenizer.go

```diff
@@ -1,15 +1,43 @@
 package v1
 
 import (
+	"fmt"
 	"unicode/utf8"
 
+	"github.com/grafana/loki/pkg/push"
 	iter "github.com/grafana/loki/v3/pkg/iter/v2"
 )
 
 const (
 	MaxRuneLen = 4
 )
 
+type StructuredMetadataTokenizer struct {
+	prefix string
+	buf    []string
+}
+
+func NewStructuredMetadataTokenizer(prefix string) *StructuredMetadataTokenizer {
+	return &StructuredMetadataTokenizer{
+		prefix: prefix,
+		buf:    make([]string, 6),
+	}
+}
+
+// Tokens implements the NGramBuilder interface
+func (t *StructuredMetadataTokenizer) Tokens(kv push.LabelAdapter) iter.Iterator[string] {
+	combined := fmt.Sprintf("%s=%s", kv.Name, kv.Value)
+	t.buf = append(t.buf[:0],
+		kv.Name,
```

Comment:
why are we adding the name and the value separately?

I think we can achieve the same with `combined`

Reply:
I would start indexing name and value for now as well. Once we see how to implement the read path, we can remove the unnecessary tokens.